swift/Workflow/Sources/WorkflowHost.swift

```diff
@@ -59,6 +60,8 @@ public final class WorkflowHost<WorkflowType: Workflow> {
             outputEventObserver.send(value: outputEvent)
         }
 
+        rootNode.enableEvents()
```

Comment:
It seems like this will get a bit complicated in the case where a rendering causes an event. This call to `enableEvents` will causes something down in the tree to dequeue its pending event, which will bubble up and recursively call `self.handle(output:)`.

That seems like it will (at the very least) result in out of order debug info (the recursive invocation will call `debugger?.didUpdate` before the outer one)

Reply:
I don't have all of this architecture in my head, but it might be nice if `enableEvents` could optionally return a `WorkflowNode<WorkflowType>.Output` value, then you could predictable handle the update inline here.

Reply:
Yes - I agree that we need to handle the debug info appropriately (so it's in the "correct" order of what happened).

I *think* reordering in `WorkflowHost` to emit the debug event before calling `enableEvents` will have the correct sequence. Basically, `enableEvents` needs to be the last thing in `handle` called, so that it is properly "tail recursive" to guarantee the right order. I'll add a test case around this, to ensure whatever is done behave correctly.

Reply:
Updated `WorkflowHost` and added a test for this. By having `enableEvents` be the last thing called in `handle` it's now ensured that the debug info comes through in the correct order.